base/MAPL_ObsUtil.F90

```diff
@@ -582,31 +576,24 @@ function get_filename_from_template_use_index (obsfile_start_time, obsfile_inter
     !
     !
     allow_wild_char=.true.
-    !
     j= index(file_template, '*')
-    if (j>0) then
-       ! wild char exist
-       !!print*, 'pos of * in template =', j
-       file_template_left = file_template(1:j-1)
-       call fill_grads_template ( filename_left, file_template_left, &
-            experiment_id='', nymd=nymd, nhms=nhms, _RC )
-       filename2= trim(filename_left)//trim(file_template(j:))
-       call fglob(filename2, filename3, rc=status)
+    if (.NOT. allow_wild_char .AND. j>0) then
+       _FAIL("* is not allowed in template")
+    end if
+    call fill_grads_template ( filename, file_template, &
+         experiment_id='', nymd=nymd, nhms=nhms, _RC )
+    if (j==0) then
+       ! exact file name
+       inquire(file= trim(filename), EXIST = exist)
+    else
+       ! now filename is:  file*.nc
+       call fglob(filename, filename2, rc=status)
```

Comment:
GitHub won't let me propose the relevant code here because some of the lines are pre-existing.   The bit just below this can be replaced with:
```
   exist = .false.
   if (status == 0) then
      exist = .true.
     filename = trim(filename2)
   end if
```

I.e., eliminate the else block.

I'd like to avoid the nested conditional but not liking the options at first glance, so we'll settle just for this.